src/test/java/io/pinecone/PineconeIndexOperationsTest.java

```diff
@@ -238,5 +243,105 @@ public void testConfigureIndex() throws IOException {
         assertEquals(expectedConfiguredIndex, configuredIndex);
         assertEquals(requestCaptor.getValue().method(), "PATCH");
         assertEquals(requestCaptor.getValue().url().toString(), "https://api.pinecone.io/indexes/testIndex");
+
+        // Test for empty string for index name
+        PineconeException thrownEmptyIndexName = assertThrows(PineconeValidationException.class, () -> client.configureIndex("",
+                configureIndexRequest));
+        assertEquals("Index name cannot be null or empty", thrownEmptyIndexName.getMessage());
+
+        // Test for null as index name
+        PineconeException thrownNullIndexName = assertThrows(PineconeValidationException.class, () -> client.configureIndex(null,
+                configureIndexRequest));
+        assertEquals("Index name cannot be null or empty", thrownNullIndexName.getMessage());
+
+        // Test for null as configureIndexRequest
+        PineconeException thrownNullRequestObj = assertThrows(PineconeValidationException.class,
+                () -> client.configureIndex("testIndex", null));
+        assertEquals("ConfigureIndexRequest object cannot be null", thrownNullRequestObj.getMessage());
+    }
+
+    @Test
+    public void testCreateCollectionWithCollectionRequestObject() throws IOException {
+        String filePath = "src/test/resources/collectionCreation.json";
+        String JsonStringCollection = new String(Files.readAllBytes(Paths.get(filePath)));
+        CollectionModel expectedCollection = gson.fromJson(JsonStringCollection, CollectionModel.class);
+
+        CreateCollectionRequest createCollectionRequest = new CreateCollectionRequest();
+
+        Call mockCall = mock(Call.class);
+        Response mockResponse = new Response.Builder()
+                .request(new Request.Builder().url("http://localhost").build())
+                .protocol(Protocol.HTTP_1_1)
+                .code(200)
+                .message("OK")
+                .body(ResponseBody.create(JsonStringCollection, MediaType.parse("application/json")))
+                .build();
+
+        when(mockCall.execute()).thenReturn(mockResponse);
+
+        OkHttpClient mockClient = mock(OkHttpClient.class);
+        when(mockClient.newCall(any(Request.class))).thenReturn(mockCall);
+        when(mockCall.execute()).thenReturn(mockResponse);
+
+        Pinecone client = new Pinecone.Builder("testAPiKey").withOkHttpClient(mockClient).build();
+        CollectionModel collection = client.createCollection(createCollectionRequest);
+
+        ArgumentCaptor<Request> requestCaptor = ArgumentCaptor.forClass(Request.class);
+
+        verify(mockClient, times(1)).newCall(requestCaptor.capture());
```

Comment:
Since we're not using the `requestCaptor` to assert or test anything, I think it can just be removed entirely. I was using it in other places in this test class to assert on what the `OkHttpClient` client is being called with after a user makes a request to `pinecone.foo()`.

Reply:
Oo okay thanks! Yeah TBH I kept it b/c I wasn't 100% sure what it was doing there and didn't want to mess w/the pattern 😓